types/defines/vectorize.d.ts

```diff
@@ -19,6 +20,27 @@ interface VectorizeError {
   error: string;
 }
 
+/**
+ * Comparison logic/operation to use for metadata filtering.
+ *
+ * This list is expected to grow as support for more operations are released.
+ */
+type VectorizeVectorMetadataFilterOp = "$eq" | "$neq";
+
+/**
+ * Filter criteria for vector metadata used to limit the retrieved query result set.
+ */
+interface VectorizeVectorMetadataFilter {
+  [field: string]:
```

Comment:
The key cannot be an operator in current implementation (a validation rejects these on the service); can this be expressed?

Reply:
Unfortunately it cannot: negated types seem to be lacking in TS support options. I tried a few workarounds, but because this is an index type we're fairly restricted in how complex these types can be.

For now, I suggest we catch on the backend